### PR TITLE
Blockchain Persist performance tweak

### PR DESCRIPTION
**Pull request by ixje**
**What current issue(s) does this address, or what feature is it adding?**
The `Persist()` call in `LevelDBBlockchain.py` loops over the inputs and unnecessary recalculates the `systemshare` every time

https://github.com/CityOfZion/neo-python/blob/ea794abbf941638fbe957d511614f3d66a7c2ef1/neo/Implementations/Blockchains/LevelDB/LevelDBBlockchain.py#L677

**How did you solve this problem?**
add a small cache
**Before**
> (venv) Eriks-Air:neo-python erik$ python -m timeit -s 'from neo.Implementations.Blockchains.LevelDB.LevelDBBlockchain import LevelDBBlockchain' 'LevelDBBlockchain.SystemShare().Hash.ToBytes()'
10000 loops, best of 3: 130 usec per loop

**After**
> (venv) Eriks-Air:neo-python erik$ python -m timeit -s 'from neo.Implementations.Blockchains.LevelDB.LevelDBBlockchain import LevelDBBlockchain' 'LevelDBBlockchain.SystemShare().Hash.ToBytes()'
100000 loops, best of 3: 3.01 usec per loop

**How did you make sure your solution works?**
`timeit` test

**Are there any special changes in the code that we should be aware of?**
no

**Please check the following, if applicable:**

- [ ] Did you add any tests?
- [X] Did you run `make lint`?
- [ ] Did you run `make test`?
- [X] Are you making a PR to a feature branch or development rather than master?
- [X] Did you add an entry to `CHANGELOG.rst`? (if not, please do)

### CODE REVIEW

File: neo/Core/Blockchain.py [hunk 1]
```diff
@@ -58,6 +59,7 @@ def StandbyValidators():
         return Blockchain.__validators
 
     @staticmethod
+    @lru_cache(maxsize=2)
```
Comment (ixje) [inline neo/Core/Blockchain.py]: fyi; `maxsize=2` is the smallest cache size configurable.

**Comment by localhuman** [comment]
That's great!

Perhaps we could make it even faster by storing a reference of `Blockchain.SystemShare().Hash.ToBytes()` ( same with SystemCoin) in LevelDBBlockchain?

**Comment by ixje** [comment]
@localhuman it turns out that's actually slower 🤔 I ran the test a couple of times and it averages around 3.45 (with the lowest time at 3.18 usec). It's just a small difference between the two solutions. 

> (venv) Eriks-Air:neo-python erik$ python -m timeit -s 'from neo.Implementations.Blockchains.LevelDB.LevelDBBlockchain import LevelDBBlockchain' 'LevelDBBlockchain.SystemShare().Hash.ToBytes()'
100000 loops, best of 3: 3.45 usec per loop

for
```
    @staticmethod
    def SystemShare():
        """
        Register AntShare.

        Returns:
            RegisterTransaction:
        """
        if not Blockchain._SystemShare:
            amount = Fixed8.FromDecimal(sum(Blockchain.GENERATION_AMOUNT) * Blockchain.DECREMENT_INTERVAL)
            owner = ECDSA.secp256r1().Curve.Infinity
            admin = Crypto.ToScriptHash(PUSHT)
            Blockchain._SystemShare = RegisterTransaction([], [], AssetType.GoverningToken,
                                                          "[{\"lang\":\"zh-CN\",\"name\":\"小蚁股\"},{\"lang\":\"en\",\"name\":\"AntShare\"}]",
                                                          amount, 0, owner, admin)
        return Blockchain._SystemShare
```

**Comment by localhuman** [comment]
I was thinking of caching/storing the actual `hash.ToBytes` value

**Comment by localhuman** [comment]
We can circle back on further performance changes, this is good for now!

**Pull request merged**